hcl2template/types.packer_config.go

```diff
@@ -28,6 +28,9 @@ type PackerConfig struct {
 	InputVariables Variables
 	LocalVariables Variables
 
+	// TODO @sylviamoss write description
+	BuildVariables Variables
```

Comment:
🤔  I don't think this is going to solve the issue because a `PackerConfig` can contain the configuration for multiple builders and this is going to cause a race conditions and all.

Reply:
A better solution would be to start a builder and then pass the builder variables to the eval context at the moment when we parse the content of the provisioners/pps etc.